### DISCUSSION TITLE
Update node from another document

### DISCUSSION THREAD

**Discussion by Hazzard17h** [mikefarah/yq#1817] [Q&A] [answered]
From documentation it's possible to [update node from another file](https://mikefarah.gitbook.io/yq/operators/assign-update#update-node-from-another-file), but it seems not possible to do the same with documents:
```shell
$ echo -e "a: apples\n---\nb: bob" | yq 'select(di==0).a = select(di==1) | select(di==0)'
a: apples
```

**Comment by mikefarah** [comment, marked as answer]
By default, `yq` runs through documents one at a time (saves on memory), you can make it evaluate against all the documents by specifying eval-all/ea:

```
echo -e "a: apples\n---\nb: bob" | yq ea 'select(di==0).a = select(di==1) | select(di==0)'
```
